src/models/activity.ts

```diff
@@ -0,0 +1,42 @@
+import {DynamoDbImage, parseStringArray} from "../services/dynamodb-images";
+import {debugLog} from "../services/logger";
+
+export interface Activity {
+    id?: string;
+    activityType?: string;
+    activityDay?: string;
```

Comment:
What is this `activityDay` please? I am not sure we are capturing this as part of the activity.
I couldn't find it in the 'dynamo' schema
```js
export const ActivitySchema = Joi.object().keys({
    parentId: Joi.string().optional(),
    activityType: Joi.any().only([ activitiesTypes ]).required(),
    testStationName: Joi.string().required(),
    testStationPNumber: Joi.string().required(),
    testStationEmail: Joi.string().email().required(),
    testStationType: Joi.any().only([ stationTypes ]).required(),
    testerName: Joi.string().min(1).max(60).required(),
    testerStaffId: Joi.string().required(),
    testerEmail: Joi.any().when("activityType", {
        is: ActivityType.VISIT,
        then:  Joi.string().email().required(),
        otherwise: Joi.any().forbidden(),
    }),
    startTime: Joi.string().optional(),
    endTime: Joi.string().optional().allow(null),
    waitReason: Joi.array().items([ waitReasons ]).optional(),
    notes: Joi.string().allow(null)
});
```

Also `testStationType` is not in there so i assume we are not interested in it?.
I can't see the `parentId` either which are required for all activities type except visit (wait and unaccountable time)

Reply:
`activityDay` is present in DynamoDB tables cvs-[integration|develop]-activities and samples have it populated, it is present in API schema. `testStationType` is present in samples too and in your code excerpt - we need it for fingerprint matching in database. Relationship/logic between attributes are not relevant for cvs-tsk-update-store to my mind (as long as it does not affect the way we store the data) - we store Activity essentially as-is.

Reply:
Thanks! I will create ticket for this so we can update our schemas and also move the api specs in the repo assuming we don't have secrets, etc.. in them.
There is probably a use case we will need to discuss with the team  as per my understanding:
An activity/visit can be of different types: visit, unaccountable time, wait time and I couldn't whether we are capturing the following relationship:
visit activity can have 'sub-activities' which can be of type wait or unaccountable time where a parentId is needed and the parentId is activityId of the visit activity.
basically when starting a visit in the mobile app this will create a visit and unaccountable time (1), wait times (2) are added if the user does nothing after a certain amount of time (2), or just close a test (1) for whatever reasons.

Reply:
@ChrisPeska-BJSS Can you answer this one?

Reply:
Thanks! That is piece of good information. I assumed there must be hierarchic relationship within data - the presence of `parentId`. We can disregard this and store data in `Activity` table only. It all depends how this data is going to be utilised. At the moment the database is hosted on AWS Aurora (MySql), which unlike other RDBMSes poorly supports hierarchic recursive queries (recursive common table expression or dedicated sytnax e.g. Oracle Connect By). It will be surmountable to query and extract hierarchic information, but it is not going to be straight-forward. I need to have @SanjeetBJSS steer on it ? Possibly, we are going to keep it as it is and store everything in one table.

Having said that should the model have 'enum' types ?

Reply:
The Activities table stores two types of data.  You get the Visit and then you get Wait time & Unaccountable time.  These last two are very similar and relate to a parent visit.  You could model these in separate tables if you want to avoid sparsely populated tables.  Otherwise, you could merge them together.

The parent relationship is only a single level.  i.e. a parent cannot have a parent.  So while you might want to join the table to itself to get visits and the wait/unaccounted information for it - you wouldn't want to tree walk.

Reply:
Need to add indexes on activityId and parentId so we can link the activities to themselves via those columns @ChrisPeska-BJSS